Sources/Bow/Effects/Typeclasses/MonadDefer.swift

```diff
@@ -0,0 +1,26 @@
+import Foundation
+
+public protocol MonadDefer : MonadError where E == Error {
+    func deferExecution<A>(_ fa : @escaping () -> Kind<F, A>) -> Kind<F, A>
+}
+
+public extension MonadDefer {
+    public func invoke<A>(_ f : @escaping () throws -> A) -> Kind<F, A> {
```

Comment:
This should be called `delay` and it will be also renamed in Arrow in 0.8.0 since the invoke operator creates a lot of ambiguity